TypeScript: Request constructor should accept Request as second parameter (per Fetch API spec)

## Summary
The generated TypeScript types for the `Request` constructor don't allow passing a `Request` object as the second parameter, even though this is valid per the [Fetch API specification](https://fetch.spec.whatwg.org/#dom-request) and works correctly at runtime.

## Current Behavior
```typescript
const original = new Request('https://example.com/old', { method: 'POST', body: 'data' });
const cloned = original.clone();

// ❌ TypeScript error TS2345:
// Argument of type 'Request' is not assignable to parameter of type 'RequestInit'
const newRequest = new Request('https://example.com/new', cloned);
```

Error:
```
error TS2345: Argument of type 'Request<CfHostMetadata, Cf>' is not assignable to parameter of type 'RequestInit<CfProperties<unknown>>'.
  Types of property 'cf' are incompatible.
    Type 'Cf | undefined' is not assignable to type 'CfProperties<unknown> | undefined'.
```

## Expected Behavior
The code should compile without error, as it does at runtime. Per the Fetch API spec, the `Request` constructor accepts either `RequestInit` or another `Request` as the second parameter.

## Spec Reference
From [Fetch Standard § 5.4](https://fetch.spec.whatwg.org/#dom-request):
> The `new Request(input, init)` constructor steps are:
> 1. Let *request* be null.
> 2. ...
> 3. If *input* is a string, then: ...
> 4. Otherwise: Set *request* to *input*'s request.
> 5. ...
> 14. If *init* is given, then: ...

The spec shows that `init` can be used alongside an existing Request to override specific properties while copying the rest.

## Use Case
This pattern is common when you need to clone a request but change its URL:

```typescript
// Rewrite URL while preserving method, headers, body, etc.
const rewrittenRequest = new Request(newUrl, originalRequest.clone());
```

## Current Workaround
```typescript
const rewrittenRequest = new Request(newUrl, originalRequest.clone() as RequestInit);
```

## Suggested Fix
Update the Request constructor type signature from:
```typescript
new <CfHostMetadata = unknown, Cf = CfProperties<CfHostMetadata>>(
    input: RequestInfo<CfProperties> | URL, 
    init?: RequestInit<Cf>
): Request<CfHostMetadata, Cf>;
```

To:
```typescript
new <CfHostMetadata = unknown, Cf = CfProperties<CfHostMetadata>>(
    input: RequestInfo<CfProperties> | URL, 
    init?: RequestInit<Cf> | Request
): Request<CfHostMetadata, Cf>;
```

## Environment
- Types generated via `wrangler types` (wrangler 4.44.0+)
- workerd runtime types from `worker-configuration.d.ts`

--- a/src/workerd/api/http.h
+++ b/src/workerd/api/http.h
@@ -824,28 +824,28 @@ class Request final: public Body {
             constructor(input: RequestInfo<CfProperties> | URL, init?: RequestInit<Cf>);
             clone(): Request<CfHostMetadata, Cf>;
             cache?: "no-store" | "no-cache" | "reload";
-            get cf(): Cf | undefined;
+            cf?: Cf;
           });
         } else if (flags.getCacheNoCache()) {
           JSG_TS_OVERRIDE(<CfHostMetadata = unknown, Cf = CfProperties<CfHostMetadata>> {
             constructor(input: RequestInfo<CfProperties> | URL, init?: RequestInit<Cf>);
             clone(): Request<CfHostMetadata, Cf>;
             cache?: "no-store" | "no-cache";
-            get cf(): Cf | undefined;
+            cf?: Cf;
           });
         } else {
           JSG_TS_OVERRIDE(<CfHostMetadata = unknown, Cf = CfProperties<CfHostMetadata>> {
             constructor(input: RequestInfo<CfProperties> | URL, init?: RequestInit<Cf>);
             clone(): Request<CfHostMetadata, Cf>;
             cache?: "no-store";
-            get cf(): Cf | undefined;
+            cf?: Cf;
           });
         }
       } else {
         JSG_TS_OVERRIDE(<CfHostMetadata = unknown, Cf = CfProperties<CfHostMetadata>> {
           constructor(input: RequestInfo<CfProperties> | URL, init?: RequestInit<Cf>);
           clone(): Request<CfHostMetadata, Cf>;
-          get cf(): Cf | undefined;
+          cf?: Cf;
         });
       }
 

--- a/src/workerd/api/tests/http-test-ts.ts
+++ b/src/workerd/api/tests/http-test-ts.ts
@@ -47,6 +47,37 @@ async function assertFetchCacheRejectsError(
   );
 }
 
+// Test that the Request constructor accepts a Request object as the second parameter.
+// This validates the TypeScript type definitions are correct.
+// Reference: https://github.com/cloudflare/workerd/issues/5940
+export const requestConstructorWithRequestAsInit = {
+  async test() {
+    // Create an original request
+    const originalRequest = new Request('https://example.com/original', {
+      method: 'POST',
+      headers: { 'X-Custom-Header': 'test-value' },
+    });
+
+    // TypeScript should allow passing a Request as the second parameter
+    const newRequest = new Request(
+      'https://example.com/new-url',
+      originalRequest
+    );
+
+    assert.strictEqual(newRequest.url, 'https://example.com/new-url');
+    assert.strictEqual(newRequest.method, 'POST');
+    assert.strictEqual(newRequest.headers.get('X-Custom-Header'), 'test-value');
+
+    // Also test with clone() which is a common pattern
+    const clonedAsInit = new Request(
+      'https://example.com/another-url',
+      originalRequest.clone()
+    );
+    assert.strictEqual(clonedAsInit.url, 'https://example.com/another-url');
+    assert.strictEqual(clonedAsInit.method, 'POST');
+  },
+};
+
 export const cacheMode = {
   async test(_ctrl: any, env: any, _ctx: any) {
     const allowedCacheModes: Set<RequestCache> = new Set([

--- a/types/generated-snapshot/experimental/index.d.ts
+++ b/types/generated-snapshot/experimental/index.d.ts
@@ -2103,7 +2103,7 @@ interface Request<
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/signal)
    */
   signal: AbortSignal;
-  cf: Cf | undefined;
+  cf?: Cf;
   /**
    * The **`integrity`** read-only property of the Request interface contains the subresource integrity value of the request.
    *

--- a/types/generated-snapshot/experimental/index.ts
+++ b/types/generated-snapshot/experimental/index.ts
@@ -2109,7 +2109,7 @@ export interface Request<
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/signal)
    */
   signal: AbortSignal;
-  cf: Cf | undefined;
+  cf?: Cf;
   /**
    * The **`integrity`** read-only property of the Request interface contains the subresource integrity value of the request.
    *

--- a/types/generated-snapshot/latest/index.d.ts
+++ b/types/generated-snapshot/latest/index.d.ts
@@ -2058,7 +2058,7 @@ interface Request<
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/signal)
    */
   signal: AbortSignal;
-  cf: Cf | undefined;
+  cf?: Cf;
   /**
    * The **`integrity`** read-only property of the Request interface contains the subresource integrity value of the request.
    *

--- a/types/generated-snapshot/latest/index.ts
+++ b/types/generated-snapshot/latest/index.ts
@@ -2064,7 +2064,7 @@ export interface Request<
    * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/signal)
    */
   signal: AbortSignal;
-  cf: Cf | undefined;
+  cf?: Cf;
   /**
    * The **`integrity`** read-only property of the Request interface contains the subresource integrity value of the request.
    *

